kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -0,0 +1,448 @@
+                    pcardDocument.getTargetAccountingLine(0).setSubAccountNumber(procurementCardHolderDetail.getSubAccountNumber());
+                    pcardDocument.getTargetAccountingLine(0).setFinancialSubObjectCode(procurementCardHolderDetail.getFinancialSubObjectCode());
+                    if ( LOG.isInfoEnabled() ) {
+                        LOG.info("Rerouting doc #" + pcardDocumentId + " at AccountFullEdit for error account Fiscal Officer.");
+                    }                        
+                    requeueDocument(pcardDocument, pcardDocumentId, node, "HasReconciler"); 
+                    // pause between reroutes
+                    try {
+                        Thread.sleep(3000);
+                    }
+                    catch (InterruptedException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+                
+              //step two, reroute documents enroute to the reconciler group(s) that had group membership problems
+                else if (node.equals("ProcurementCardReconciler") && groupRerouting && 
+						parameterService.getParameter(ProcurementCardRerouteDocumentsStep.class,
+								RECONCILER_GROUPS_TO_REROUTE_PARM_NM).getValue() != null
+						&& parameterService
+								.getParameter(ProcurementCardRerouteDocumentsStep.class,
+										RECONCILER_GROUPS_TO_REROUTE_PARM_NM)
+								.getValue().contains(procurementCardHolderDetail.getReconcilerGroupId())) {
+                    if ( LOG.isInfoEnabled() ) {
+						LOG.info("Rerouting doc #" + pcardDocumentId + " at ProcurementCardReconciler, in "
+								+ RECONCILER_GROUPS_TO_REROUTE_PARM_NM + " group "
+								+ procurementCardHolderDetail.getReconcilerGroupId());
+                    }
+                    requeueDocument(pcardDocument, pcardDocumentId, node, "ProcurementCardReconciler"); 
+                    // pause between reroutes
+                    try {
+                        Thread.sleep(3000);
+                    }
+                    catch (InterruptedException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+                
+            }                            
+        }
+
+        return true;
+    }
+    
+    /**
+     * Gets the default Chart Code, Account from the custom Procurement Cardholder table.
+     * 
+     */
+    private ProcurementCardDefault getProcurementCardDefault(String creditCardNumber) {
+                
+        Map<String, String> pkMap = new HashMap<String, String>();
+        pkMap.put("creditCardNumber", creditCardNumber);
+        ProcurementCardDefault procurementCardDefault = (ProcurementCardDefault) businessObjectService.findByPrimaryKey(ProcurementCardDefault.class, pkMap);
```

Comment:
I'm not sold on this... if a getter should be used as though the businessObjectService was being fetched from an outside class, what's the point of extending the bases class and having the visibility of businessObjectService opened up to 'protected'?